nx.json

```diff
@@ -0,0 +1,39 @@
+{
+  "tasksRunnerOptions": {
+    "default": {
+      "runner": "nx/tasks-runners/default",
+      "options": {
+        "cacheableOperations": ["test:lib"]
+      }
+    }
+  },
+  "defaultBase": "main",
+  "namedInputs": {
+    "globalBuildAffectingConfig": [
+      "{workspaceRoot}/babel.config.js",
+      "{workspaceRoot}/rollup.config.js",
+      "{workspaceRoot}/rollup.config.ts",
+      "{workspaceRoot}/tsconfig.json",
+      "{workspaceRoot}/tsconfig.base.json"
+    ],
+    "globalNonBuildAffectingConfig": [
+      "{workspaceRoot}/.eslintrc",
+      "{workspaceRoot}/jest-preset.js"
+    ],
+    "default": ["{projectRoot}/**/*", "globalBuildAffectingConfig", "globalNonBuildAffectingConfig"],
+    "public": [
+      "default",
+      "!{workspaceRoot}/.eslintrc",
+      "!{workspaceRoot}/jest-preset.js",
+      "!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap)",
+      "!{projectRoot}/.eslintrc",
+      "!{projectRoot}/jest.config.js"
```

Comment:
will this have to change on `alpha` where we are not using `jest` anymore?

I think, again, maybe it would be better to just introduce those changes on `alpha` ?

Reply:
Most of these would stay the same - we'd really just want to change out the `jest-preset.js` and `jest.config.js` files for `vitest.config.ts` and `test-setup.ts`.

The surface area here in is quite small, which is why I thought to adjust in `main` directly.

Happy to submit to `alpha` if that's preferred though! Just let me know which you'd prefer :)

Reply:
okay let's keep it on main, and I'll merge it back to alpha and solve the conflicts